From b36390e49601065ec0786c54d22dd190e23d67f3 Mon Sep 17 00:00:00 2001
From: Nullifidian <rick.romig@gmail.com>
Date: Sun, 30 Apr 2023 11:17:30 -0400
Subject: [PATCH] syncrick 3.0.6: Changed variable declaration for local_host.
 Added check for client hostname to prevent syncing of .git directory from the
 laptop.

---
 syncrick | 18 ++++++++++++------
 1 file changed, 12 insertions(+), 6 deletions(-)

diff --git a/syncrick b/syncrick
index e8e1ac4..9cb02f1 100755
--- a/syncrick
+++ b/syncrick
@@ -26,12 +26,12 @@ fi
 
 ## Variables ##
 
-local_host="$(uname -n)"; readonly local_host
 _script=$(basename "$0"); readonly _script
-readonly _version="3.0.5"
-readonly _updated="26 Apr 2023"
+readonly _version="3.0.6"
+readonly _updated="30 Apr 2023"
 readonly conf_dir="$HOME/.config/scripts"
 readonly conf_file="$_script.conf"
+readonly local_host="${HOSTNAME:-$(hostname)}"
 
 ## Functions ##
 
@@ -109,9 +109,15 @@ sync_script() {
 
   # Sync repositories
   echo $'\n'$"${orange}Syncing Gitea repositoies...${normal}"
-  rsync -avzh --delete /home/"$USER"/gitea/ "$localnet.$hostip":gitea/
-  exit_status="$?"
-  transfer_status "$exit_status"
+  if [[ "$local_host" == "$lap_host" ]]; then
+    rsync -avzh --delete --exclude '.git' /home/"$USER"/gitea/ "$localnet.$hostip":gitea/
+    exit_status="$?"
+    transfer_status "$exit_status"
+  else
+    rsync -avzh --delete /home/"$USER"/gitea/ "$localnet.$hostip":gitea/
+    exit_status="$?"
+    transfer_status "$exit_status"
+  fi
 
   # Sync personal files.
   echo $'\n'$"${orange}Syncing Audio Books...${normal}"
-- 
2.34.1

